api/v1beta1/glanceapi_types.go

```diff
@@ -107,6 +107,43 @@ type GlanceAPISpec struct {
 	// +kubebuilder:validation:Optional
 	// ExtraMounts containing conf files and credentials
 	ExtraMounts []GlanceExtraVolMounts `json:"extraMounts"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkAttachments is a list of NetworkAttachment resource names to expose the services to the given network
+	NetworkAttachments []string `json:"networkAttachments"`
+
+	// +kubebuilder:validation:Optional
+	// ExternalEndpoints, expose a VIP via MetalLB on the pre-created address pool
+	ExternalEndpoints []MetalLBConfig `json:"externalEndpoints"`
+}
+
+// MetalLBConfig to configure the MetalLB loadbalancer service
+type MetalLBConfig struct {
```

Comment:
Is this something we're going to replicate on all the operators that require the MetalLB info [1]? It's technically ok but I'm wondering if we want to have this struct as part of `lib-common` instead of re-implementing for each operator. Also, I think this struct is not going to change over the time (or diverge according to specific operator's needs), so I'm proposing to have a common definition.

[1] https://github.com/openstack-k8s-operators/cinder-operator/pull/136

Reply:
we plan to move this into the infra-operator when we add there more network related stuff, like ipam, dns and so to have it versioned there within the api. until then we keep it in the operators

Reply:
ack and thanks for clarifying, I agree to get this information from `infra-operator` as more (common) network related components are going to be defined there.